env/work/srw_python/srwlib.py

```diff
@@ -4271,6 +4272,19 @@ def srwl_wfr_emit_prop_multi_e(_e_beam, _mag, _mesh, _sr_meth, _sr_rel_prec, _n_
         #iAvgProc += 1 #OC190414 (commented-out)
         #iSave += 1
 
+    #MR01112016: Initialize prarameters for the SRW status files and generate the files:
+    log_dir = os.getcwd() if _file_path is None else os.path.dirname(os.path.abspath(_file_path))
+    timestamp = '{:%Y-%m-%d_%H-%M-%S}'.format(datetime.datetime.now())
+    log_file = 'srw_status_{}'.format(timestamp)
+    log_path = os.path.join(log_dir, log_file)
+    if nProc <= 1:
+        total_num_of_particles = nPartPerProc
+    else:
+        total_num_of_particles = nSentPerProc * (nProc - 1) * _n_part_avg_proc
+    if rank == 0:
+        srwl_uti_save_stat_wfr_emit_prop_multi_e(0, total_num_of_particles, filename=log_path,
+                                                 cores=nProc, particles_per_iteration=_n_part_avg_proc)
+
```

Comment:
Oleg, I improved the logging feature, so that it now:
- works in both cases - with MPI and without it;
- saves status files to the data directory along with the `.dat` files (if the directory is not specified, it saves the files to the current directory);
- has a name in format of `srw_status_2016-11-02_09-27-00.json` and `srw_status_2016-11-02_09-27-00.log`, so all the files should have unique logs, and there should not be any conflicts if you run separate parallel calculations.

For more details please see https://github.com/mrakitin/SRW/issues/18 and the related commit https://github.com/mrakitin/SRW/commit/ecb76540236f01271c653b6dc91929365b7e1102.